### PR TITLE
feat(platform): ability to add rows to table

### LINKED ISSUE #7422: [Feature Request] Add new line into a table - editable row
#### Is this a bug, enhancement, or feature request?
Feature request

#### Briefly describe your proposal.
We have a [table](https://fundamental-ngx.netlify.app/#/core/table#table-toolbar) in which we want to add a line.
We saw that in the actual component there is a '+' that open a popup.
From Fiori we saw that there are three possible ways to [add a new line into a table](https://experience.sap.com/internal/fiori-design-web/grid-table/#add-items).
We are interested in having the first one:
"Add the item inline. Create an empty, editable row as the first item of the table. Show the Save button on the table toolbar. This option is recommended for simple scenarios with just a few columns and no option to hide columns."

Thanks in advance

#### Which versions of Angular and Fundamental Library for Angular are affected? (If this is a feature request, use current version.)
Angular 13.0.2
Fundamental 0.33.0-rc.21

### LINKED ISSUE #7538: bug(platform): cannot enter space in custom table column
#### Is this a bug, enhancement, or feature request?
bug
#### Briefly describe your proposal.
In Custom Column example there's an input field for the description column. If user tries to enter space, it's being ignored.
#### Which versions of Angular and Fundamental Library for Angular are affected? (If this is a feature request, use current version.)
current
#### If this is a bug, please provide steps for reproducing it.
1. Go to https://fundamental-ngx.netlify.app/#/platform/table#custom-columnshttps://fundamental-ngx.netlify.app/#/platform/table#custom-columns
2. Try to type some text containing space into a custom column for the description.
ER: Space should be inserted into input field
AR: Space is ignored.

### PR DESCRIPTION
**Pull request by N1XUS**
## Related Issue(s)

closes #7422 closes #7538

## Description
Added ability to add new rows to the platform table. Added default controls which can be overriten with custom templates.
## Screenshots
image

##### During Implementation

1. Visual Testing:

-   [x] visual misalignments/updates
-   [x] check Light/Dark/HCB/HCW themes
-   [x] RTL/LTR - proper rendering and labeling
-   [x] responsiveness(resize)
-   [x] Content Density (Cozy/Compact/(Condensed))
-   [x] States - hover/disabled/focused/active/on click/selected/selected hover/press state
-   [x] Interaction/Animation - open/close, expand/collapse, add/remove, check/uncheck
-   [x] Mouse vs. Keyboard support
-   [x] Text Truncation

2. API and functional correctness

-   [x] check for console logs (warnings, errors)
-   [x] API boundary values
-   [x] different combinations of components - free style
-   [x] change the API values during testing

3. Documentation and Example validations

-   [x] missing API documentation or it is not understandable
-   [x] poor examples
-   [x] Stackblitz works for all examples

4. Accessibility testing
5. Browser Testing - Edge, Safari, Chrome, Firefox

##### PR Quality

-   [x] the commit message(s) follows the guideline:
        https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
-   [x] tests for the changes that have been done
-   [x] all items on the PR Review Checklist are addressed :
        https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist
-   [x] Run npm run build-pack-library and test in external application
-   [x] update `README.md`
-   [x] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)

### CODE REVIEW

File: libs/platform/src/lib/table/components/table-column/table-column.component.ts [hunk 1]
```diff
@@ -146,6 +149,11 @@ export class TableColumnComponent extends TableColumn implements OnInit, OnChang
         this.columnCellTemplate = fdpCellDef?.templateRef;
     }
 
+    @ContentChild(FdpEditableCellDef)
```
Comment (InnaAtanasova) [inline libs/platform/src/lib/table/components/table-column/table-column.component.ts:152]: typedocs

**Review by InnaAtanasova — APPROVED** [review]

**Review by platon-rov — APPROVED** [review]

**Review by mikerodonnell89 — CHANGES_REQUESTED** [review]
1. If I click "Create" then immediately "Cancel", the save/cancel buttons are still visible, I think they should be hidden 
2. If I create a new row and save it, then create another new row and save it, that first one I created disappears from the table

**Comment by N1XUS** [comment]
> 1. If I click "Create" then immediately "Cancel", the save/cancel buttons are still visible, I think they should be hidden
> 2. If I create a new row and save it, then create another new row and save it, that first one I created disappears from the table
> 3. Any way to make cells required / disabling "Save" button if required fields missing?

Hey Mike. Your comments has been addressed

**Review by mikerodonnell89 — CHANGES_REQUESTED** [review]
One more thing I just noticed in this example is searching for something that turns up 0 rows, then resetting the search, the table remains empty

**Comment by N1XUS** [comment]
@mikerodonnell89 data source has been updated. Please re-check

**Review by droshev — APPROVED** [review]

**Pull request merged**